tutorials/11-probabilistic-pca/11_probabilistic-pca.jmd

```diff
@@ -213,400 +352,80 @@ You can find more details about this in the Bishop book mentioned in the introdu
-for i in 1:n_samples
-    U_ni = orthogonal_matrix(D, Q, vv_mat[i, :, :])
-    W_ni = U_ni * (LinearAlgebra.I(Q) .* sigma[:, i])
-    z_n = W_ni' * transpose(Matrix(dat))
-    samples_raw[:, :, i] = z_n
-end
+df_pro = DataFrame(z[alpha_indices, :]', :auto)
+rename!(df_pro, Symbol.(["z" * string(i) for i in collect(1:k)]))
+df_pro[!, :cell] = 1:n_cells
 
-# initialize a 3D plot with 1 empty series
-plt = plot(
-    [100, 200, 300];
-    xlim=(-4.00, 7.00),
-    ylim=(-100.00, 0.00),
-    group=["Setosa", "Versicolor", "Virginica"],
-    markercolor=["red", "blue", "black"],
-    title="Visualization",
-    seriestype=:scatter,
+df_pro[!, :type] = repeat([1, 2]; inner=n_cells ÷ 2)
+scatter(
+    df_pro[:, 1], df_pro[:, 2]; xlabel="z1", ylabel="z2", color=df_pro[:, "type"], label=""
 )
-
-anim = @animate for i in 1:n_samples
-    scatter!(
-        plt,
-        samples_raw[1, 1:50, i],
-        samples_raw[2, 1:50, i];
-        color="red",
-        seriesalpha=0.1,
-        label="",
-    )
-    scatter!(
-        plt,
-        samples_raw[1, 51:100, i],
-        samples_raw[2, 51:100, i];
-        color="blue",
-        seriesalpha=0.1,
-        label="",
-    )
-    scatter!(
-        plt,
-        samples_raw[1, 101:150, i],
-        samples_raw[2, 101:150, i];
-        color="black",
-        seriesalpha=0.1,
-        label="",
-    )
-end
-gif(anim, "anim_fps.gif"; fps=5)
-```
-
-Here we see the density of the projections from the chain to illustrate the uncertainty in the
-projections. We can see quite clearly that it is possible to separate Setosa from Versicolor and
-Virginica species, but that the latter two cannot be clearly separated based on the pPCA projection. This can be
-shown even more clearly by using a kernel density estimate and plotting the contours of that estimate.
-
-```julia; echo=false
-let
-    m1 = mean(samples_raw[1, 1:50, :])
-    m2 = mean(samples_raw[1, 51:100, :])
-    m3 = mean(samples_raw[1, 101:150, :])
-    @assert m1 - m2 > 3
-    @assert m1 - m3 > 3
-    @assert m2 - m3 < 2
-end
-```
-
-```julia
-# kernel density estimate
-using KernelDensity
-dens = kde((vec(samples_raw[1, :, :]), vec(samples_raw[2, :, :])))
-StatsPlots.plot(dens)
 ```
 
-```julia, echo=false, skip="notebook", tangle=false
-if isdefined(Main, :TuringTutorials)
-    Main.TuringTutorials.tutorial_footer(WEAVE_ARGS[:folder], WEAVE_ARGS[:file])
-end
 ```
+This plot is very similar to the low-dimensional plot above, with the *relevant* dimensions chosen based on the values of $\alpha$ via ARD. When you are in doubt about the number of dimensions to project onto, ARD might provide an answer to that question.
+
+# Final comments.
+p-PCA is a linear map which linearly transforms the data between the original and projected spaces.
+It can also thought as a matrix factorisation method, in which $x=(w \times z)^T$. The projection matrix can be understood as a new basis in the projected space, and $z$ are the new coordinates.
```

Comment:
```suggestion
It can also thought as a matrix factorisation method, in which $\mathbf{X}=(\mathbf{W} \times \mathbf{Z})^T$. The projection matrix can be understood as a new basis in the projected space, and $\mathbf{Z}$ are the new coordinates.
```